### PR TITLE
executor: remove unused structure `RecordBatch`

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Remove unused structure `RecordBatch`.

### What is changed and how it works?
Only refactor, no logical change.

### Check List 

Tests 

 - No code

Code changes

 - Has interface methods change

### CODE REVIEW

File: util/sqlexec/restricted_sql_executor.go [hunk 1]
```diff
@@ -87,10 +87,10 @@ type RecordSet interface {
 	Fields() []*ast.ResultField
 
 	// Next reads records into chunk.
-	Next(ctx context.Context, req *chunk.RecordBatch) error
+	Next(ctx context.Context, req *chunk.Chunk) error
 
-	//NewRecordBatch create a recordBatch.
-	NewRecordBatch() *chunk.RecordBatch
+	//NewFirstChunk create a chunk.
+	NewFirstChunk() *chunk.Chunk
```
Comment (ngaut) [inline util/sqlexec/restricted_sql_executor.go]: s/NewFirstChunk/NewChunk/

Comment (qw4990) [reply util/sqlexec/restricted_sql_executor.go]: It has been renamed.

File: executor/aggregate.go [hunk 2]
```diff
@@ -512,17 +513,20 @@ func (w *HashAggFinalWorker) run(ctx sessionctx.Context, waitGroup *sync.WaitGro
 	w.getFinalResult(ctx)
 }
 
-// Next implements the Executor Next interface.
-func (e *HashAggExec) Next(ctx context.Context, req *chunk.RecordBatch) error {
+func (e *HashAggExec) Next(ctx context.Context, req *chunk.Chunk) error {
+	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
```
Comment (zz-jason) [inline executor/aggregate.go]: can be removed? see https://github.com/pingcap/tidb/pull/10875

File: executor/bind.go [hunk 3]
```diff
@@ -38,7 +39,12 @@ type SQLBindExec struct {
 }
 
 // Next implements the Executor Next interface.
-func (e *SQLBindExec) Next(ctx context.Context, req *chunk.RecordBatch) error {
+func (e *SQLBindExec) Next(ctx context.Context, req *chunk.Chunk) error {
+	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
```
Comment (zz-jason) [inline executor/bind.go]: ditto

File: executor/aggregate.go (hunk 2 above)
Comment (qw4990) [reply executor/aggregate.go]: My mistake.
The unnecessary code has been removed.

**Comment by qw4990** [comment]
> Please resolve the conflicts

Resolved.
PTAL~

**Comment by qw4990** [comment]
Waiting for https://github.com/pingcap/tidb/pull/10941;

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**